monitor.cpp

```diff
@@ -3,36 +3,43 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <string>
+using std::cout, std::endl, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
+void PrintMessage(const std::string& message) {
+  cout << message << endl;
+  for (int i = 0; i < 6; i++) {
+    cout << "\r* " << flush;
+    sleep_for(seconds(1));
+    cout << "\r *" << flush;
+    sleep_for(seconds(1));
+  }
+}
+
+int CheckBodyTemperature(float bodyTemperature) {
+  if (bodyTemperature > 102 || bodyTemperature < 95) {
+    PrintMessage("Temperature is critical!");
     return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
+  }
+  return 1;
+}
+
+int CheckPulseRate(float pulseRate) {
+  if (pulseRate < 60 || pulseRate > 100) {
```

Comment:
Can further create a function to check if the parameter is in range. Like `parameterIsInRange(float inputValue, float lowerLimit, float upperLimit)`
Then you can test for all ranges only once, no need to add three new tests for every parameter added.